engines/sync.c

```diff
@@ -149,20 +148,14 @@ static int fio_pvsyncio2_queue(struct thread_data *td, struct io_u *io_u)
     {
         if ((o->hipri_percentage) && (io_u->ddir == DDIR_READ))
         {
-            if (++o->hipri_count % 100/o->hipri_percentage)
-            {
+            if ((rand()%100 <= o->hipri_percentage) == 0) {
                 dprint(FD_IO, "Disable RWF_HIPRI\n");
-            }
-            else
-            {
+            } else {
                 dprint(FD_IO, "Enable RWF_HIPRI\n");
                 flags |= RWF_HIPRI;
             } 
```

Comment:
I think we would prefer this to use fio's own internal random number generator routines rather than `rand()` to make things deterministic...